### PR TITLE
fix #62

### LINKED ISSUE #62: Fails if $HOME/.mycroft doesn't exist
Currently the saving of skills.json doesn't validate that the folder `~/.mycroft` folder exists.

The [write_skill_state()](https://github.com/MycroftAI/mycroft-skills-manager/blob/master/msm/skill_state.py#L24) function should validate that the directory exists and try to create it if it doesn't.

### PR DESCRIPTION
**Pull request by flozi00**
fix https://github.com/MycroftAI/mycroft-skills-manager/issues/62#issue-547414079

### CODE REVIEW

**Comment by krisgesling** [comment]
Hey Florian, there's some other interesting methods available in `os` that might be useful here.

Maybe checkout: 
https://docs.python.org/3.5/library/os.path.html#os.path.isdir
https://docs.python.org/3.5/library/os.html?highlight=makedirs#os.makedirs

**Comment by flozi00** [comment]
Hey Kris, ~/ is not working with this functions.
Shell command creates directory if not already existing.
The validation is built in by linux

**Comment by forslund** [comment]
You can use the expanduser function (already used in this file) to get the full path

File: msm/skill_state.py [hunk 1]
```diff
@@ -22,6 +23,10 @@ def load_device_skill_state() -> dict:
 
 
 def write_device_skill_state(data: dict):
+    try:
+        os.system("mkdir ~/.mycroft")
```
Comment (forslund) [inline msm/skill_state.py]: I suggest you get the directory from the SKILL_STATE_PATH global,

```python
dir_path = dirname(expanduser(SKILL_STATE_PATH))
```

For the creating maybe pathlib's mkdir or os.mkdirs could be used?

**Comment by flozi00** [comment]
better this time ?

**Comment by forslund** [comment]
Is there a reason you're not using `os.path.dirname` to get the dirname from the original path?

**Comment by flozi00** [comment]
I just didn't thought about

**Pull request closed without merging**